HW4.py

```diff
@@ -0,0 +1,360 @@
+import os
+import sys
+
+import numpy as np
+import pandas as pd
+import scipy.stats as ss
+import sklearn.datasets
+import statsmodels.api
+from plotly import express as px
+from plotly import figure_factory as ff
+from plotly import graph_objects as go
+from scipy.stats import binned_statistic
+from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
+from sklearn.inspection import permutation_importance
+from sklearn.metrics import confusion_matrix
+
+
+def classify_type(data):
```

Comment:
[This](https://stackoverflow.com/questions/35826912/what-is-a-good-heuristic-to-detect-if-a-column-in-a-pandas-dataframe-is-categori) gives a nice heuristic for this.